File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        char[] sarr = s.toCharArray();
        char[] tarr = t.toCharArray();

        HashMap<Character, Integer> shm = new HashMap<>();
        HashMap<Character, Integer> thm = new HashMap<>();

        for (char ch : sarr) {
            if (shm.containsKey(ch)) {
                shm.put(ch, shm.get(ch) + 1);
            }
            else {
                shm.put(ch, 1);
            }
        }

        for (char ch : tarr) {
            if (thm.containsKey(ch)) {
                thm.put(ch, thm.get(ch) + 1);
            }
            else {
                thm.put(ch, 1);
            }
        }

        if (shm.equals(thm)) {
            return true;
        }
        else {
            return false;
        }
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hms = {}
        hmt = {}
        for ch in s:
            if ch in hms:
                hms[ch] += 1
            else:
                hms[ch] = 1
        
        for ch in t:
            if ch in hmt:
                hmt[ch] += 1
            else:
                hmt[ch] = 1
        
        for key, value in hms.items():
            if key not in hmt:
                return False
            else:
                if value == hmt[key]:
                    del hmt[key]
        
        if len(hmt) == 0:
            return True
        
        return False